### DISCUSSION TITLE
How to select all?

### DISCUSSION THREAD

**Discussion by donutheist** [wezterm/wezterm#2026] [Q&A]
On my Mac, I am used to CMD-A selecting all the text in the window. I don't seem to be able to find a way to do that in Wezterm. Am I missing something?

**Comment by wez** [comment]
The closest thing at the moment is to [Copy Mode](https://wezfurlong.org/wezterm/copymode.html) and do something like this:

* `CTRL-SHIFT-X` - to enter copy mode
* `H` - to move to top of the viewport
* `v` - to start selecting
* `G` - to move to the bottom of the viewport
* `CTRL-SHIFT-C` to copy to the clipboard
* `Esc` to exit copy mode.

**Reply by MuhammedZakir** [reply]
Additionally:
1. If you want to copy the entire scrollback buffer, use `g` instead of `H` in the above step.
2. To do this with just one key-combo, use `Multiple` -- https://wezfurlong.org/wezterm/config/lua/keyassignment/Multiple.html.

**Comment by piechologist** [comment]
You can add a custom key binding to your `wezterm.lua`:

```lua
{ key = 'a', mods = 'CMD', action = wezterm.action_callback(function(window, pane)
    local dims = pane:get_dimensions()
    local txt = pane:get_text_from_region(0, dims.scrollback_top, 0, dims.scrollback_top + dims.scrollback_rows)
    window:copy_to_clipboard(txt:match('^%s*(.-)%s*$')) -- trim leading and trailing whitespace
    end)
},
```

**Reply by eugenesvk** [reply]
Thanks! As an extra pro tip: this is a much more flexible solution, and allows skipping of the bottom rows (by adding an offset to the last argument) in case your shell supports it (but only for those shells that support it! via parsing `pane:get_foreground_process_info()` fields to check which shell is in the foreground)

**Comment by donutheist** [comment]
I found this, which is a little easier
```
        {
            key =  "a",
            mods = "SUPER",
            action = wezterm.action_callback(function(window, pane)
                local selected = pane:get_lines_as_text(pane:get_dimensions().scrollback_rows)
                window:copy_to_clipboard(selected, 'Clipboard')
            end)
        },
```

**Comment by Raytle** [comment]
At https://wezterm.org/config/files.html#configuration-file-structure why not include a link to this issue as well as an example such as the following…

-- Pull in the wezterm API
`local wezterm = require 'wezterm'`

-- This will hold the configuration.
`local config = wezterm.config_builder()`

-- Set the font to JetBrains Mono with size 14 pt
`config.font = wezterm.font 'JetBrains Mono'`
`config.font_size = 14.0`

-- For example, changing the color scheme:
`config.color_scheme = 'AdventureTime'`

-- Keybinding to select all text and copy it to the clipboard using Ctrl+A
```config.keys = {
  {
    key = 'a',
    mods = 'CTRL',  -- Use CTRL for Ctrl+A
    action = wezterm.action_callback(function(window, pane)
      local dims = pane:get_dimensions()
      local txt = pane:get_text_from_region(0, dims.scrollback_top, 0, dims.scrollback_top + dims.scrollback_rows)
      window:copy_to_clipboard(txt:match('^%s*(.-)%s*$'))  -- Trim leading and trailing whitespace
    end)
  },
}
```
-- and finally, return the configuration to wezterm
return config

**Comment by jcrben** [comment]
Feature request issue at https://github.com/wezterm/wezterm/issues/5094

**Reply by Raytle** [reply]
### I use Ctrl+A to select all

dozens of times each week.

[Do I stutter](https://github.com/wezterm/wezterm/discussions/2026#discussioncomment-12676361)? I repeat...

-- Keybinding to select all text and copy it to the clipboard using Ctrl+A
```config.keys = {
  {
    key = 'a',
    mods = 'CTRL',  -- Use CTRL for Ctrl+A
    action = wezterm.action_callback(function(window, pane)
      local dims = pane:get_dimensions()
      local txt = pane:get_text_from_region(0, dims.scrollback_top, 0, dims.scrollback_top + dims.scrollback_rows)
      window:copy_to_clipboard(txt:match('^%s*(.-)%s*$'))  -- Trim leading and trailing whitespace
    end)
  },
}
```
The maintainer is remiss. He should put this information in the documentation as I previously requested. 

What? Oh. Gee. Sorry. No. I don't know how to perform pull requests. I don't use Git.
